pkg/apiserver/archive/archive.go

```diff
@@ -53,7 +57,9 @@ type StatusResponse struct {
 // Register mounts our HTTP handler on the mux.
 func Register(r *gin.RouterGroup, s *Service) {
 	endpoint := r.Group("/archives")
-	endpoint.Use(utils.AuthRequired)
+	endpoint.Use(func(c *gin.Context) {
+		utils.AuthRequired(c, s.conf.ClusterScoped, s.conf.TargetNamespace)
```

Comment:
Should we bump the auth middleware to the top of the group? This can let us set the `namespace` query in advance to omit subsequent processing.

For example:

```go
q := c.Request.URL.Query()
q.Add("test", "testtest")
c.Request.URL.RawQuery = q.Encode()

test := c.Query("test") // testtest
```

Reply:
I try to update the query in `gin.Context`, but I found that it not work. Then I read the source code of gin, I find that there is a `queryCache` in context, and when we call `context.Query("text")` it is in fact read data from the cache, and I don't find some functions to update this cache 

https://github.com/gin-gonic/gin/blob/ee4de846a894e9049321e809d69f4343f62d2862/context.go#L447

Reply:
😢 After taking a look at the code, it seems that this cache is not very useful.

I think there should be a `Reset` method to reset the cache. So, we can solve this problem in the future once gin has the feature of cache reset.